ibu-imager/ops/ops.go

```diff
@@ -67,3 +79,85 @@ func (o *ops) RunBashInHostNamespace(command string, args ...string) (string, er
 	args = append([]string{command}, args...)
 	return o.RunInHostNamespace("bash", "-c", strings.Join(args, " "))
 }
+
+// WaitForEtcd waits for the availability of an etcd server at the given endpoint.
+// It continuously checks for the existence of a specific key in etcd to determine the server's availability.
+// If the etcd server becomes accessible, it returns nil, indicating success. Otherwise, it returns an error.
+func (o *ops) WaitForEtcd(endpoint string) error {
+	// Create an etcd client configuration with the specified endpoint and timeout
+	etcdConfig := client.Config{
+		Endpoints:   []string{endpoint},
+		DialTimeout: 5 * time.Second,
+	}
+
+	// Initialize the etcd client
+	etcdClient, err := client.New(etcdConfig)
+	if err != nil {
+		return err
+	}
+	defer etcdClient.Close()
+
+	// Create a context with a timeout to limit the waiting duration
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Continuously attempt to retrieve a specific key from etcd
+	for {
+		_, err = etcdClient.Get(ctx, "some_key")
```

Comment:
etcd has a healthz endpoint, no need for checking for keys, which also means we can drop the etcd client dependency

Reply:
+1

Reply:
+1